[CUMULUS-2251] distribution module bug (#1989)
nasa/cumulus: CHANGELOG.md

### Migration notes
- Update the name of your `cumulus_message_adapter_lambda_layer_arn` variable for the `cumulus` module to `cumulus_message_adapter_lambda_layer_version_arn`. The value of the variable should remain the same (a layer version ARN of a Lambda layer for the [`cumulus-message-adapter`](https://github.com/nasa/cumulus-message-adapter/).
- **CUMULUS-2138** - Update all workflows using the `MoveGranules` step to add `UpdateGranulesCmrMetadataFileLinksStep`that runs after it. See the example [`IngestAndPublishWorkflow`](https://github.com/nasa/cumulus/blob/master/example/cumulus-tf/ingest_and_publish_granule_workflow.asl.json) for reference.
- **CUMULUS-2251**

>   - Add new resource defintion `egress_api_gateway_log_subscription_filter` to `cumulus-tf/main.tf`.

Reviewer: ```suggestion
  - Because it has been removed from the `cumulus` module, a new resource definition for `egress_api_gateway_log_subscription_filter` must be added to `cumulus-tf/main.tf`. For reference on how to define this resource, see [`example/cumulus-tf/main.tf`](https://github.com/nasa/cumulus/blob/master/example/cumulus-tf/main.tf) .
```

  - Adds ability to filter granule inventory reports
- **CUMULUS-2211**
  - Adds `granules/bulkReingest` endpoint to `@cumulus/api`
- **CUMULUS-2251**

>   - Adds `log_api_gateway_to_cloudwatch` variable to `cumulus-tf/variables.tf`

Reviewer: ```suggestion
  - Adds `log_api_gateway_to_cloudwatch` variable to `example/cumulus-tf/variables.tf`
```

- **CUMULUS-2201**
  - Update `dbIndexer` lambda to process requests in serial
  - Fixes ingestPdrWithNodeNameSpec parsePdr provider error
- **CUMULUS-2251**

>   - Move Egress Api Gateway Log Group Filter from `distribution/main.tf` to `/cumulus-tf/main.tf` and remove `tea_api_egress_log_group` variable from `distribution/variables.tf` and `cumulus/variables.tf` and add `log_api_gateway_to_cloudwatch` variable to `cumulus-tf/variables.tf`

Reviewer: It would be good to split these notes about what was changed, removed, and added into separate notes under the different headers for Changed, Added, etc.

Reviewer: We also need to add a migration note here - since people will be expected to add the new resource definition to their cumulus-tf/main.tf file. Also, we need to update the [template-deploy repo](https://github.com/nasa/cumulus-template-deploy/pulls) that we maintain as a starter/example for users with these changes.

>   - This fixes the count error users ran into when they had `log_api_gateway_to_cloudwatch` set to true in their `thin_egress_app` module.

Reviewer: We also usually add a separate header for "Fixed" that you could put this under. I would also maybe adjust the language to "fixed a deployment error caused by depending on a module output for a resource count" since "count error" is a bit ambiguous without more context

- **CUMULUS-2201**
  - Update `dbIndexer` lambda to process requests in serial
  - Fixes ingestPdrWithNodeNameSpec parsePdr provider error
- **CUMULUS-2251**

>   - Moves Egress Api Gateway Log Group Filter from `distribution/main.tf` to `/cumulus-tf/main.tf` and remove `tea_api_egress_log_group` variable from `distribution/variables.tf` and `cumulus/variables.tf`.

Reviewer: Nitpicking for sure, but let's move the "removed.." statements under a "Removed" header

- **CUMULUS-2201**
  - Update `dbIndexer` lambda to process requests in serial
  - Fixes ingestPdrWithNodeNameSpec parsePdr provider error
- **CUMULUS-2251**

>   - Moves Egress Api Gateway Log Group Filter from `distribution/main.tf` to `/cumulus-tf/main.tf`

Reviewer: ```suggestion
  - Moves Egress Api Gateway Log Group Filter from `tf-modules/distribution/main.tf` to `example/cumulus-tf/main.tf`
```

### Fixed
- **CUMULUS-2251**
  - This fixes a deployment error caused by depending on the `thin_egress_app` module output for a resource count.

### Removed
- **CUMULUS-2251**

>   -  Removes `tea_api_egress_log_group` variable from `distribution/variables.tf` and `cumulus/variables.tf`.

Reviewer: ```suggestion
  -  Removes `tea_api_egress_log_group` variable from `tf-modules/distribution/variables.tf` and `tf-modules/cumulus/variables.tf`.
```
